src/destreamer.ts

```diff
@@ -347,6 +347,31 @@ async function downloadVideo(videoGUIDs: Array<string>,
         logger.info(`Video no.${videos.indexOf(video) + 1} downloaded!!\n\n`);
     }
 
+    // START CONCATENATE
+    if(argv.concat){
+        logger.info("Concatenating videos...");
+        let concatFiles = "";
+        let concatName = `videos_concatenated ${videos.length}`;
+        for(const video of videos){
+            concatFiles += ("file '" + video.outPath + "'\n");
+            logger.info("file " + video.outPath);
+        }
+
+        // Create a file containg all videos path
+        await fs.promises.writeFile('files.txt', concatFiles);
+        
+        const concatCommand = (
+            // set video concat settings
+            `ffmpeg -f concat -safe 0 -protocol_whitelist file,http,https,tcp,tls,crypto ` +
+            // add video list
+            `-i files.txt ` +
+            // copy codec and output path
+            `-c copy videos/${concatName}.mkv`
+        );
+        execSync(concatCommand, {stdio: 'ignore'}); //TODO: remove video files after they have been concateated
```

Comment:
Instead of using FFmpeg you could concatenate the video by literally concatenating the file byte per byte.
You could take a look at pipes in node to handle file redirection, if your up to the task.
If not this way is perfectly fine and I'll approve, let me know.

Reply:
I will look at this and see if I can do it.

Reply:
Wait, no! What are we concatenating here, segments or entire finished videos? If the latter, you can't simply concatenate bytes together since every .mkv or .mp4 file has its own header with track/codec information so you absolutely positively need ffmpeg to handle the concat as it will require reencoding almost every time.

Reply:
We are concatenating finished videos, but at the end of the concatenation the code won't delete single video files.